elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreHelper.java

```diff
@@ -261,4 +268,43 @@ private void validateHavingClause(FilterExpression havingClause) {
             validateHavingClause(((NotFilterExpression) havingClause).getNegated());
         }
     }
+
+    /**
+     * Method to verify that all the sorting options provided
+     * by the user are valid and supported.
+     */
+    public void validateSorting() {
+        Sorting sorting = entityProjection.getSorting();
+        if (sorting == null) {
+            return;
+        }
+        Set<String> allFields = getRelationships();
+        allFields.addAll(getAttributes());
+        Map<Path, Sorting.SortOrder> sortClauses = sorting.getValidSortingRules(entityProjection.getType(), dictionary);
+        sortClauses.keySet().forEach((path) -> validatePath(path, allFields));
+    }
+
+    /**
+     * Verifies that the current path can be sorted on
+     * @param path The path that we are validating
+     * @param allFields Set of all field names included in initial query
+     */
+    private void validatePath(Path path, Set<String> allFields) {
+        List<Path.PathElement> pathElemenets = path.getPathElements();
+
+        //TODO add support for double nested sorting
+        if (pathElemenets.size() > 2) {
+            throw new UnsupportedOperationException(
+                    "Currently sorting on double nested fields is not supported");
+        }
+        Path.PathElement currentElement = pathElemenets.get(0);
+        String currentField = currentElement.getFieldName();
+        Class<?> currentClass = currentElement.getType();
+        if (!allFields.stream().anyMatch(field -> field.equals(currentField))) {
+            throw new InvalidOperationException("Can't sort on field that is not present in query");
+        }
+        if (dictionary.getIdFieldName(currentClass).equals(currentField) || currentField.equals("id")) {
```

Comment:
I think there is a `REGULAR_ID_NAME` or something like this that represents the reserved `id` string. Let's replace `id` with that.